### PR TITLE
Bumping versions to 1.4

### PR DESCRIPTION
**Pull request by robertbastian**

### CODE REVIEW

**Review by sffc — APPROVED** [review]

**Review by sffc — CHANGES_REQUESTED** [review]
Blocking 1.4 until we achieve agreement on https://github.com/unicode-org/icu4x/pull/4321

File: provider/baked/compactdecimal/Cargo.toml [hunk 1]
```diff
@@ -15,3 +14,4 @@ homepage.workspace = true
 include.workspace = true
 repository.workspace = true
 rust-version.workspace = true
+version.workspace = true
```
[the remark this answers is not in the document]
Comment (robertbastian) [reply provider/baked/compactdecimal/Cargo.toml]: yes this file is a mistake

**Comment by robertbastian** [comment]
https://github.com/unicode-org/icu4x/pull/4320

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,8 @@
 
 ## Unreleased
```
[the remark this answers is not in the document]
Comment (robertbastian) [reply CHANGELOG.md:3]: I'd rather do this than have the next person add it? Also it's nice to see that nothing is unreleased, everything is published!

**Review by sffc — APPROVED** [review]
Un-blocking the release.

File: CHANGELOG.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (sffc) [reply CHANGELOG.md:3]: It looks confusing as if everything below it is unreleased. It's easy to overlook the levels of nesting.

**Pull request merged**